nixos/modules/config/users-groups.nix

```diff
@@ -931,6 +931,18 @@ in {
       }
     ] ++ flatten (flip mapAttrsToList cfg.users (name: user:
       [
+        (
+          let
+            # Things fail in various ways with especially non-ascii usernames.
+            # This regex mirrors the one from shadow's is_valid_name:
```

Comment:
Should we add a short comment why the trailing $ was removed?

Reply:
Done!